internal/syscall.go

```diff
@@ -226,3 +226,20 @@ type wrappedErrno struct {
 func (we wrappedErrno) Unwrap() error {
 	return we.Errno
 }
+
+type syscallError struct {
+	error
+	errno syscall.Errno
+}
+
+func SyscallError(err error, errno syscall.Errno) error {
+	return &syscallError{err, errno}
+}
```

Comment:
Why not export the type and use it directly? This is what golang stdlib does with types such as `os.PathError`.

Reply:
I mean, instead of e.g `return syscall.SyscallError(err, errno)` you can do `return &syscall.SyscallError{err, errno}` and avoid an unnecessary function call (not sure if golang will inline it). Since this is an internal type, you're not exporting it.